### PR TITLE
support customized attibutes and aria-* attributes

### PR DESCRIPTION
**Pull request by chilijung**
https://facebook.github.io/react/docs/jsx-gotchas.html#custom-html-attributes

Customized attributes like `data-custom-attributes` are converted into camel case, it should not be converted. So does `aria-*` attributes.

### CODE REVIEW

**Comment by Olical** [comment]
A perfect PR, thanks a lot! I like the look of the changes and the tests, it makes sense to me. If React has this edge case supported then so should we.

File: src/Element.js [hunk 1]
```diff
@@ -88,7 +88,13 @@ Element.prototype.attributeNameMappings = {
 }
 
 Element.prototype.attributeToPropName = function (name) {
-  return this.attributeNameMappings[name] || camelCase(name)
+  if (name.match(/^data-/)) {
```
Comment (Olical) [inline src/Element.js]: I forgot to mention before I merged it that these RegExps should probably be initialised in a closure (cached and reused).

Comment (chilijung) [inline src/Element.js]: OK, I'll send another PR if I have time to fix it. Thanks :+1:

**Comment by Olical** [comment]
I opened an issue as a reminder, I'll get around to it if not but feel free! :) thanks.

**Comment by Olical** [comment]
Published in v2.1.0
